In [1]:
 import pandas as pd
df = pd.read_csv("allbacks.csv")

In [2]:
df.head()

,rownames,volume,area,weight,cover
0,1,885,382,800,hb
1,2,1016,468,950,hb
2,3,1125,387,1050,hb
3,4,239,371,350,hb
4,5,701,371,750,hb


In [3]:
df.shape

(15, 5)

In [4]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 15 entries, 0 to 14
Data columns (total 5 columns):
 #   Column    Non-Null Count  Dtype 
---  ------    --------------  ----- 
 0   rownames  15 non-null     int64 
 1   volume    15 non-null     int64 
 2   area      15 non-null     int64 
 3   weight    15 non-null     int64 
 4   cover     15 non-null     object
dtypes: int64(4), object(1)
memory usage: 732.0+ bytes


In [5]:
df.describe()

,rownames,volume,area,weight
count,15.000000,15.000000,15.000000,15.000000
mean,8.000000,845.266667,182.800000,706.666667
std,4.472136,339.879909,203.586837,268.837887
min,1.000000,239.000000,0.000000,250.000000
25%,4.500000,618.000000,0.000000,512.500000
50%,8.000000,929.000000,0.000000,725.000000
75%,11.500000,1025.000000,376.500000,950.000000
max,15.000000,1492.000000,468.000000,1075.000000


## Intial Observations
I noticed that although the count for area values is 15 the min,25%,50%,75% are all 0. This seems unsual since in the textbook the values show positive area measurements. 

In [6]:
df["cover"]

0     hb
1     hb
2     hb
3     hb
4     hb
5     hb
6     hb
7     pb
8     pb
9     pb
10    pb
11    pb
12    pb
13    pb
14    pb
Name: cover, dtype: object

The median of the cover type is paperback (pb). There are 15 books, so the median is the 8th value after sorting which is pb.

In [7]:
print(df["weight"].mean())

706.6666666666666


The mean weight for all books is 706.67 grams. This represents the average weight of the 15 books and is close to the median weight of 725 grams.

In [8]:
print(df.groupby("cover")["weight"].mean())

cover
hb    796.428571
pb    628.125000
Name: weight, dtype: float64


The mean weight for hardback books is 796.43 grams and paperback books is 628.13 grams. Hardback books are 168.3 grams heavier on avergae than paperback books. Based on these results, the library may want to choose paperback books in the future since they weigh less.

## Verification of paperback average weight by hand

In [9]:
df

,rownames,volume,area,weight,cover
0,1,885,382,800,hb
1,2,1016,468,950,hb
2,3,1125,387,1050,hb
3,4,239,371,350,hb
4,5,701,371,750,hb
5,6,641,367,600,hb
6,7,1228,396,1075,hb
7,8,412,0,250,pb
8,9,953,0,700,pb
9,10,929,0,650,pb


The weights for pb are : 250,700,650,975,350,950,425,725

(250+700+650+975+350+950+425+725)/8

5025/8=628.13

The mean paperback weight calculated by hand is 628.13 grams, which matches the result calculated using Python.



https://github.com/jazzminowens/cs82a-portfolio